From 5f5ba9da420ef5539b96ab198610785008307e1f Mon Sep 17 00:00:00 2001
From: Max Ihlenfeldt <max@igalia.com>
Date: Mon, 21 Oct 2024 10:08:17 +0000
Subject: [PATCH] Revert "Allow (and use)
 std::hardware_destructive_interference_size."

libcxx only defines `std::hardware_{con,de}structive_interference_size`
if `__GCC_{CON,DE}STRUCTIVE_SIZE` are defined. clang only defines these
since version 19.1, see
https://github.com/llvm/llvm-project/commit/72c373bf.

As Scarthgap uses clang 18.1, we need to revert
https://crrev.com/c/5767325, which uses
`std::hardware_destructive_interference_size`.

Upstream-Status: Inappropriate [specific to older versions of clang]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
[Rebase on 136.0.7103.16: keep only source code change for enum_table.h]
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 .../media_router/common/providers/cast/channel/enum_table.h   | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/components/media_router/common/providers/cast/channel/enum_table.h b/components/media_router/common/providers/cast/channel/enum_table.h
index fdf8941c32..9340ba2857 100644
--- a/components/media_router/common/providers/cast/channel/enum_table.h
+++ b/components/media_router/common/providers/cast/channel/enum_table.h
@@ -12,7 +12,6 @@
 
 #include <cstdint>
 #include <cstring>
-#include <new>
 #include <optional>
 #include <ostream>
 #include <string_view>
@@ -368,7 +367,8 @@ class EnumTable {
 
  private:
 #ifdef ARCH_CPU_64_BITS
-  alignas(std::hardware_destructive_interference_size)
+  // Align the data on a cache line boundary.
+  alignas(64)
 #endif
       std::initializer_list<Entry> data_;
   bool is_sorted_;
